assets/userCtrl.js

```diff
@@ -285,13 +289,20 @@ module.exports = function ($scope, $location, io) {
       var game = $scope.user.flairGames[j];
       if (game.ign) {
         hasIGN = true;
+        if (game.ign.length > 12) {
```

Comment:
So the way to solve that is to have 2 functions. 1 that takes the igns,
fcs, tsvs etc and actually does the work, and 1 that takes the string and
tries to break it down to the components and pass that on to the other
function.

But alright, this is too much work for this pr. I'll add another issue and
merge this.

On Tue, 8 Dec 2015, 09:37 not-an-aardvark <notifications@github.com> wrote:

> In assets/userCtrl.js
> <https://github.com/pokemontrades/flairhq/pull/453#discussion_r46931818>:
>
> > @@ -285,13 +289,20 @@ module.exports = function ($scope, $location, io) {
> >        var game = $scope.user.flairGames[j];
> >        if (game.ign) {
> >          hasIGN = true;
> > +        if (game.ign.length > 12) {
>
> We could do this, but we might have to sacrifice some specificity. The
> client has explicit fields for FCs/IGNs/games, whereas the server has to
> extract the info from a (possibly-malformed) flair text. So the client is
> able to say "this game contains an illegal character" whereas the server is
> sometimes only be able to say, "there's a formatting error but I'm not sure
> what it is" because the illegal character messes up the parsing.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pokemontrades/flairhq/pull/453/files#r46931818>.
>